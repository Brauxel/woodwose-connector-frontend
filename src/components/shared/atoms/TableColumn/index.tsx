import React from 'react'
import styled from 'styled-components'

interface TableColumnComponentProps {
  borderWidth?: string
  borderStyle?: string
  borderColor?: string
}

interface TableRowColumnProps {
  children: React.ReactNode
  styles?: TableColumnComponentProps
}

const TableColumnComponent = styled.td<TableColumnComponentProps>`
  border-width: ${({ borderWidth }) => borderWidth || '1px'};
  border-style: ${({ borderStyle }) => borderStyle || 'solid'};
  border-color: ${({ borderColor }) => borderColor || 'black'};
`

export const TableColumn: React.FC<TableRowColumnProps> = ({
  children,
  styles,
}) => <TableColumnComponent {...styles}>{children}</TableColumnComponent>
