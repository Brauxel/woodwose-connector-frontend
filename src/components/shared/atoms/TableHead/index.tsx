import React, { ThHTMLAttributes } from 'react'
import styled from 'styled-components'

interface TableHeadComponentProps {
  borderWidth?: string
  borderStyle?: string
  borderColor?: string
}

interface TableHeadProps extends ThHTMLAttributes<HTMLElement> {
  children: React.ReactNode
  styles?: TableHeadComponentProps
}

const TableHeadComponent = styled.th<TableHeadComponentProps>`
  border-width: ${({ borderWidth }) => borderWidth || '1px'};
  border-style: ${({ borderStyle }) => borderStyle || 'solid'};
  border-color: ${({ borderColor }) => borderColor || 'black'};
  text-transform: capitalize;
`

export const TableHead: React.FC<TableHeadProps> = ({
  children,
  styles,
  ...props
}) => (
  <TableHeadComponent {...props} {...styles}>
    {children}
  </TableHeadComponent>
)
