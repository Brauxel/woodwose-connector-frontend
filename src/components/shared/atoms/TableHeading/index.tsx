import React, { ThHTMLAttributes } from 'react'
import styled from 'styled-components'

interface TableHeadComponentProps {
  borderWidth?: string
  borderStyle?: string
  borderColor?: string
}

interface TableHeadingProps extends ThHTMLAttributes<HTMLElement> {
  children: React.ReactNode
  styles?: TableHeadComponentProps
}

const TableHeadingComponent = styled.th<TableHeadComponentProps>`
  border-width: ${({ borderWidth }) => borderWidth || '1px'};
  border-style: ${({ borderStyle }) => borderStyle || 'solid'};
  border-color: ${({ borderColor }) => borderColor || 'black'};
  text-transform: capitalize;
`

export const TableHeading: React.FC<TableHeadingProps> = ({
  children,
  styles,
  ...props
}) => (
  <TableHeadingComponent {...props} {...styles}>
    {children}
  </TableHeadingComponent>
)
