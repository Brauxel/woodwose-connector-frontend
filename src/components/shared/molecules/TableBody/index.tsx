import React from 'react'

interface TableBodyProps {
  children: React.ReactNode
}

export const TableBody: React.FC<TableBodyProps> = ({ children }) => (
  <tbody>{children}</tbody>
)
