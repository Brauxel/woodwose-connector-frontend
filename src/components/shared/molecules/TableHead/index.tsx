import React from 'react'

interface TableHeadProps {
  children: React.ReactNode
}

export const TableHead: React.FC<TableHeadProps> = ({ children }) => (
  <thead>{children}</thead>
)
