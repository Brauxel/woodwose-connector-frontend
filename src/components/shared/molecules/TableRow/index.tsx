import React from 'react'

interface TableRowProps {
  children: React.ReactNode
}

export const TableRow: React.FC<TableRowProps> = ({ children }) => (
  <tr>{children}</tr>
)
