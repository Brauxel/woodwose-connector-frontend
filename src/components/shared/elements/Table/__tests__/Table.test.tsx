import reactRenderer from 'react-test-renderer'
import { render } from '@testing-library/react'
import { Table } from '..'
import { TableRow } from '../../../molecules/TableRow'
import { TableColumn } from '../../../atoms/TableColumn'

describe('Table Component', () => {
  describe('render component', () => {
    it('should render the provided child JSX components', () => {
      const tableInstance = reactRenderer.create(
        <Table>
          <TableRow>
            <TableColumn>This is a table column</TableColumn>
          </TableRow>
        </Table>
      ).root

      expect(tableInstance.findByType(TableRow)).toBeTruthy()
      expect(tableInstance.findByType(TableColumn)).toBeTruthy()
    })
  })

  describe('snapshots', () => {
    it('snapshots should match when default styles are used', () => {
      const tableComponent = reactRenderer.create(
        <Table>
          <TableRow>
            <TableColumn>This is a table column</TableColumn>
          </TableRow>
        </Table>
      )

      expect(tableComponent.toJSON()).toMatchSnapshot()
    })
  })
})
