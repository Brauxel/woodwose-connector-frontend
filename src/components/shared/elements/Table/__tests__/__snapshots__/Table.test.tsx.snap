// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Table Component snapshots snapshots should match when custom styles are used 1`] = `
<table
  className="sc-bczRLJ cKCjRN"
  data-testid="table-component"
  width="80%"
>
  <tr>
    <td
      className="sc-gsnTZi ibvJHk"
    >
      This is a table column
    </td>
  </tr>
</table>
`;

exports[`Table Component snapshots snapshots should match when default styles are used 1`] = `
<table
  className="sc-bczRLJ hwejrA"
  data-testid="table-component"
>
  <tr>
    <td
      className="sc-gsnTZi ibvJHk"
    >
      This is a table column
    </td>
  </tr>
</table>
`;
